core/src/main/java/org/jivesoftware/sparkimpl/certificates/SparkTrustManager.java

```diff
@@ -187,6 +188,17 @@ public void checkServerTrusted(X509Certificate[] chain, String authType) throws
     }
 
     /**
+     * Return true if the certificate chain contain only one Self Signed certificate
+     */
+    private boolean isSelfSigned(X509Certificate[] chain) {
```

Comment:
This code assumes that the first certificate is the self-signed one, which could lead to several problems. For example, a certificate chain should, but is not always ordered (and if it is, I think it's typically the *last* certificate that's self-signed). 

Perhaps you should consider changing the argument to this method: instead of passing a chain, you might want to pass a single certificate. That way, this method's responsibility is very well defined. If needed, you could add additional methods that check if the entire chain represents a self-signed certificate (by checking if the chain size is equal to 1, and by passing that certificate to this method).